### PR TITLE
Build a Node.js client for Origami Repo Data

### PR DESCRIPTION
**Pull request by rowanmanning**
This is now a fully functioning client which implements all of the
endpoints in the Origami Repo Data service. I've included documentation
generation and have tested this against the live service.

(because of this, anything in the docs folder should be ignored when
reviewing this PR)

I need to write a suite of unit tests, but I'm going to commit these
separately. Also later on I may move the documentation generation and
npm publishing into the shared Makefile.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+		"no-irregular-whitespace": 1,
+		"no-multi-spaces": 2,
+		"one-var": [2, "never"],
+		"constructor-super": 2,
+		"no-this-before-super": 2,
+		"no-var": 2,
+		"prefer-const": 1,
+		"no-const-assign": 2
+	},
+	"globals": {
```
Comment (JakeChampion) [inline .eslintrc.js]: An alternative would be to have another eslintrc file inside the tests directory which contains:
```js
module.exports = {
    env: {
        mocha: true
    }
};
```

File: README.md [hunk 2]
```diff
@@ -2,3 +2,74 @@
...
+  - [Contributing](#contributing)
+  - [Publishing](#publishing)
+  - [Support and Migration](#support-and-migration)
+  - [Contact](#contact)
+  - [Licence](#licence)
+
+
+## Usage
+
+Running the Origami Service module requires [Node.js] 6.x and [npm]. You can install with:
```
Comment (JakeChampion) [inline README.md]: Should this say "Running the Origami Repo Data Client"?

File: docs/index.html [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width,initial-scale=1">
+    <title>Home - Documentation</title>
+
+    <script src="scripts/prettify/prettify.js"></script>
+    <script src="scripts/prettify/lang-css.js"></script>
+    <!--[if lt IE 9]>
+      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
+    <![endif]-->
+    <link type="text/css" rel="stylesheet" href="https://code.ionicframework.com/ionicons/2.0.1/css/ionicons.min.css">
```
Comment (JakeChampion) [inline docs/index.html]: What do these docs look like? They look to not use Origami :-(

File: .eslintrc.js (hunk 1 above)
Comment (rowanmanning) [reply .eslintrc.js]: Ah yep, I just copy-pasted this for now from another repo. Alright if we leave until we have a single point of truth for service/node module linting?

File: README.md (hunk 2 above)
Comment (rowanmanning) [reply README.md]: Yes! Oops

File: docs/index.html (hunk 3 above)
Comment (rowanmanning) [reply docs/index.html]: It's just an existing theme for jsdoc. I'm not adverse to eventually building a theme that uses Origami, that'd be pretty cool, but quite a big job I think

File: lib/client.js [hunk 4]
```diff
@@ -0,0 +1,361 @@
...
+	/**
+	 * Create an Origami Repo Data client.
+	 * @param {Object} [options] - The client options.
+	 * @param {String} [options.apiKey] - The API key to use when making requests.
+	 * Defaults to the value of the <code>REPO_DATA_API_KEY</code> environment variable.
+	 * @param {String} [options.apiSecret] - The API secret to use when making requests.
+	 * Defaults to the value of the <code>REPO_DATA_API_SECRET</code> environment variable.
+	 * @param {String} [options.apiUrl] - The URL of the Origami Repo Data service.
+	 * Defaults to the value of the <code>REPO_DATA_API_URL</code> environment variable or the production service.
+	 */
```
Comment (JakeChampion) [inline lib/client.js]: Missing the [returns jsdoc tag](http://usejsdoc.org/tags-returns.html).

Comment (rowanmanning) [reply lib/client.js]: `@returns` is not required for class `constructor` methods, as you can never change it from an instance of the class

File: README.md (hunk 2 above)
Comment (rowanmanning) [reply README.md]: Fixed

File: docs/index.html (hunk 3 above)
Comment (rowanmanning) [reply docs/index.html]: It looks like this:
screen shot 2018-01-15 at 12 53 38

File: lib/client.js [hunk 5]
```diff
@@ -0,0 +1,361 @@
...
+			let error;
+			if (response.body.validation) {
+				error = new Error(`${response.body.message}: ${response.body.validation.join(', ')}`);
+			} else {
+				error = new Error(response.body.message);
+			}
+			error.status = response.statusCode;
+			throw error;
+		}
+		return response.body || undefined;
```
Comment (JakeChampion) [inline lib/client.js]: I'm not sure if this is intentional but if the response body was an empty string, this function would return `undefined`.

File: .eslintrc.js (hunk 1 above)
Comment (JakeChampion) [reply .eslintrc.js]: Yep

File: docs/index.html (hunk 3 above)
Comment (JakeChampion) [reply docs/index.html]: Deal 👍

File: lib/client.js (hunk 5 above)
Comment (rowanmanning) [reply lib/client.js]: Yep, that's intentional. Only the `DELETE` methods return an empty body, and I think `undefined` makes more semantic sense in the client

File: lib/client.js (hunk 4 above)
Comment (JakeChampion) [reply lib/client.js]: It is something not many codebases do but you can change it by providing an explicit return value.

>When the code new Foo(...) is executed, the following things happen:
>
>1. A new object is created, inheriting from Foo.prototype.
>2. The constructor function Foo is called with the specified arguments, and with this bound to the newly created object. new Foo is equivalent to new Foo(), i.e. if no argument list is specified, Foo is called without arguments.
>3. The object returned by the constructor function becomes the result of the whole new expression. If the constructor function doesn't explicitly return an object, the object created in step 1 is used instead. (Normally constructors don't return a value, but they can choose to do so if they want to override the normal object creation process.)
> -- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/new#Description

Comment (rowanmanning) [reply lib/client.js]: Cool, have added

**Pull request merged**